### PR TITLE
Add finance example

### PR DESCRIPTION
**Pull request by rcchen**
Adding @jtlan's finance example to the website.

### CODE REVIEW

File: examples/finance/finance.js [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+xAxis.formatter(Plottable.Formatters.multiTime());
+var yScale = new Plottable.Scales.Linear();
+var yAxis = new Plottable.Axes.Numeric(yScale, "left");
+var colorScale = new Plottable.Scales.Color();
+
+var series1 = new Plottable.Dataset(makeSeriesData(1000), { name: "series1" });
+var series2 = new Plottable.Dataset(makeSeriesData(1000), { name: "series2" });
+
+var plot = new Plottable.Plots.Line(xScale, yScale);
+plot.x(function(d) { return d["x"]; }, xScale).y(function(d) { return d["y"]; }, yScale);
```
Comment (aicioara) [inline examples/finance/finance.js]: `d.x` instead of `d["x"]`. Same for y. Less characters

File: examples/finance/finance.js [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+plot.attr("stroke", function(d, i, dataset) { return dataset.metadata().name; }, colorScale);
+plot.addDataset(series1);
+plot.addDataset(series2);
```
Comment (aicioara) [inline examples/finance/finance.js]: Collapse the 2 addDataset into one using fluent style

File: examples/finance/finance.js [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+plot.autorangeMode("y");
+
+var sparklineXScale = new Plottable.Scales.Time();
+var sparklineXAxis = new Plottable.Axes.Time(sparklineXScale, "bottom");
+var sparklineYScale = new Plottable.Scales.Linear();
+var sparkline = new Plottable.Plots.Line(xScale, sparklineYScale);
+sparkline.x(function(d) { return d["x"]; }, sparklineXScale).y(function(d) { return d["y"]; }, sparklineYScale);
```
Comment (aicioara) [inline examples/finance/finance.js]: same

File: examples/finance/finance.js [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+sparkline.attr("stroke", function(d, i, dataset) { return dataset.metadata().name; }, colorScale);
+sparkline.addDataset(series1);
+sparkline.addDataset(series2);
```
Comment (aicioara) [inline examples/finance/finance.js]: same

File: examples/finance/finance.js [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+var series2 = new Plottable.Dataset(makeSeriesData(1000), { name: "series2" });
+
+var plot = new Plottable.Plots.Line(xScale, yScale);
+plot.x(function(d) { return d["x"]; }, xScale).y(function(d) { return d["y"]; }, yScale);
+plot.attr("stroke", function(d, i, dataset) { return dataset.metadata().name; }, colorScale);
+plot.addDataset(series1);
+plot.addDataset(series2);
+plot.autorangeMode("y");
+
+var sparklineXScale = new Plottable.Scales.Time();
```
Comment (aicioara) [inline examples/finance/finance.js]: What is a sparkline? Maybe add some comments to explain what is going on between sections? Something like
```typescript
// Creating the bottom plot
```

File: examples/finance/finance.js [hunk 6]
```diff
@@ -0,0 +1,126 @@
...
+dragBox.onDragEnd(function(bounds) {
+  if (bounds.topLeft.x === bounds.bottomRight.x) {
+    xScale.domain(sparklineXScale.domain());
+  }
+});
+xScale.onUpdate(function() {
+  dragBox.boxVisible(true);
+  var xDomain = xScale.domain();
+  dragBox.bounds({
+    topLeft: { x: sparklineXScale.scale(xDomain[0]), y: null},
```
Comment (aicioara) [inline examples/finance/finance.js]: whitespace before `}`. To be consistent with http://plottablejs.org/examples/baseball/

File: examples/finance/finance.js [hunk 7]
```diff
@@ -0,0 +1,126 @@
...
+    bottomRight: { x: sparklineXScale.scale(xDomain[1]), y: null}
+  });
+});
+var miniChart = new Plottable.Components.Group([sparkline, dragBox]);
+
+var pzi = new Plottable.Interactions.PanZoom(xScale);
```
Comment (aicioara) [inline examples/finance/finance.js]: `PanZoom(xScale, null)`

File: examples/finance/finance.js [hunk 8]
```diff
@@ -0,0 +1,126 @@
...
+
+var output = d3.select("#output");
+var outputDefaultText = "Closest:"
+output.text(outputDefaultText);
+
+var chart = new Plottable.Components.Table([
+  [yAxis, plot         ],
+  [null,  xAxis        ],
+  [null,  miniChart    ],
+  [null, sparklineXAxis]
```
Comment (aicioara) [inline examples/finance/finance.js]: align

File: examples/finance/finance.js [hunk 9]
```diff
@@ -0,0 +1,126 @@
...
+    topLeft: { x: sparklineXScale.scale(xDomain[0]), y: null},
+    bottomRight: { x: sparklineXScale.scale(xDomain[1]), y: null}
+  });
+});
+var miniChart = new Plottable.Components.Group([sparkline, dragBox]);
+
+var pzi = new Plottable.Interactions.PanZoom(xScale);
+pzi.attachTo(plot);
+
+var output = d3.select("#output");
```
Comment (aicioara) [inline examples/finance/finance.js]: The name of this div is too generic. I believe it is the outputTitle or similar?

File: examples/finance/finance.js [hunk 10]
```diff
@@ -0,0 +1,126 @@
...
+  var datum = nearestEntity.datum;
+  output.text("Closest: (" + datum.x.toLocaleString() + ", " + datum.y.toFixed(2) + ")");
+});
+pointer.onPointerExit(function(p) {
+  crosshair.hide();
+  output.text(outputDefaultText);
+});
+pointer.attachTo(plot);
+
+
```
Comment (aicioara) [inline examples/finance/finance.js]: Single `\n`

File: examples/finance/finance.js [hunk 11]
```diff
@@ -0,0 +1,126 @@
...
+function makeSeriesData(n, startDate) {
+  startDate = startDate || new Date();
+  var startYear = startDate.getUTCFullYear();
+  var startMonth = startDate.getUTCMonth();
+  var startDay = startDate.getUTCDate();
+  var toReturn = new Array(n);
+  for (var i = 0; i < n; i++) {
```
Comment (aicioara) [inline examples/finance/finance.js]: I prefer
```typescript
Array.apply(null, Array(n)).map(function(d, i) {
  return {
    x: new Date(Date.UTC(startYear, startMonth, startDay + i)),
    y: i > 0 ? toReturn[i-1].y + Math.random()*2-1 : Math.random() * 5
  }
})

File: examples/finance/finance.js [hunk 12]
```diff
@@ -0,0 +1,126 @@
...
+        toReturn[i] = {
```
Comment (aicioara) [inline examples/finance/finance.js]: Indentation

File: examples/finance/finance.js [hunk 13]
```diff
@@ -0,0 +1,126 @@
...
+            x: new Date(Date.UTC(startYear, startMonth, startDay + i)),
+            y: i > 0 ? toReturn[i-1].y + Math.random()*2-1 : Math.random() * 5
```
Comment (aicioara) [inline examples/finance/finance.js]: whitespace around `*` and `-`

File: examples/finance/finance.js [hunk 14]
```diff
@@ -0,0 +1,126 @@
...
+        };
+    };
+  return toReturn;
+}
+
+function createCrosshair(plot) {
+  var crosshair = {};
+  var crosshairContainer = plot.foreground().append("g").style("visibility", "hidden");
+  crosshair.vLine = crosshairContainer.append("line")
+                       .attr({
```
Comment (aicioara) [inline examples/finance/finance.js]: Indent

File: examples/finance/finance.js [hunk 15]
```diff
@@ -0,0 +1,126 @@
...
+      cx: p.x,
+      cy: p.y
+    });
+    crosshairContainer.style("visibility", "visible");
+  }
+  crosshair.hide = function() {
+    crosshairContainer.style("visibility", "hidden");
+  }
+  return crosshair;
+}
```
Comment (aicioara) [inline examples/finance/finance.js]: `\n`

Comment (rcchen) [inline examples/finance/finance.js]: Git won't pick up on the `\n` I typed for some reason..

Comment (aicioara) [inline examples/finance/finance.js]: It picked it up. It's fine

File: examples/finance/finance.js [hunk 16]
```diff
@@ -0,0 +1,125 @@
...
+
+function createCrosshair(plot) {
+  var crosshair = {};
+  var crosshairContainer = plot.foreground().append("g").style("visibility", "hidden");
+  crosshair.vLine = crosshairContainer.append("line")
+                    .attr({
+                      "stroke": "black",
+                      "y1": 0,
+                      "y2": plot.height()
```
Comment (aicioara) [inline examples/finance/finance.js]: Let's be consistent with the style bellow and one attribute at a time. (or the other way round and amend the bellow instead), I'm in favor of the former

File: examples/finance/finance.js [hunk 17]
```diff
@@ -0,0 +1,125 @@
...
+var series2 = new Plottable.Dataset(makeSeriesData(1000), { name: "series2" });
+
+var plot = new Plottable.Plots.Line(xScale, yScale);
+plot.x(function(d) { return d.x; }, xScale).y(function(d) { return d.y; }, yScale);
+plot.attr("stroke", function(d, i, dataset) { return dataset.metadata().name; }, colorScale);
+plot.addDataset(series1).addDataset(series2);
+plot.autorangeMode("y");
+
+// Create the sparkline at the bottom of the chart
+// For more info, see https://en.wikipedia.org/wiki/Sparkline
```
Comment (aicioara) [inline examples/finance/finance.js]: I actually think it is a bit silly now especially since it is just this single comment. Let's remove it.

**Pull request merged**